2024-03-11 22:36:08,679 AutoFE.py:35 __init__ INFO | Namespace(cuda='0', epochs=300, ppo_epochs=20, episodes=2, file_name='SPECTF', mode='classify', model='xgb', metric='f1', seed=1, train_size=0.7, cv=5, shuffle=False, d_model=128, d_k=32, d_v=32, d_ff=64, n_heads=6, worker=1, steps_num=3, lr=0.0001, entropy_weight=0.0001, baseline_weight=0.95, gama=0.9, gae_lambda=0.95, batch_size=4, c_columns=['V0', 'V1', 'V2', 'V3', 'V4', 'V5', 'V6', 'V7', 'V8', 'V9', 'V10', 'V11', 'V12', 'V13', 'V14', 'V15', 'V16', 'V17', 'V18', 'V19', 'V20', 'V21', 'V22', 'V23', 'V24', 'V25', 'V26', 'V27', 'V28', 'V29', 'V30', 'V31', 'V32', 'V33', 'V34', 'V35', 'V36', 'V37', 'V38', 'V39', 'V40', 'V41', 'V42', 'V43'], d_columns=[], target='label')
2024-03-11 22:36:08,679 AutoFE.py:36 __init__ INFO | File name: SPECTF
2024-03-11 22:36:08,679 AutoFE.py:37 __init__ INFO | Data shape: (267, 45)
2024-03-11 22:36:09,150 AutoFE.py:90 set_cuda INFO | Use device: 0, cuda:0, NVIDIA GeForce RTX 4060 Laptop GPU
2024-03-11 22:36:09,150 AutoFE.py:80 __init__ INFO | Done AutoFE initialization.
2024-03-11 22:36:09,355 AutoFE.py:103 fit_attention INFO | score_b=0.7976939203354297, scores_b=[0.83333333 0.77777778 0.77358491 0.75471698 0.8490566 ]
2024-03-11 22:36:57,734 AutoFE.py:35 __init__ INFO | Namespace(cuda='0', epochs=300, ppo_epochs=20, episodes=2, file_name='SPECTF', mode='classify', model='xgb', metric='f1', seed=1, train_size=0.7, cv=5, shuffle=False, d_model=128, d_k=32, d_v=32, d_ff=64, n_heads=6, worker=1, steps_num=3, lr=0.0001, entropy_weight=0.0001, baseline_weight=0.95, gama=0.9, gae_lambda=0.95, batch_size=4, c_columns=['V0', 'V1', 'V2', 'V3', 'V4', 'V5', 'V6', 'V7', 'V8', 'V9', 'V10', 'V11', 'V12', 'V13', 'V14', 'V15', 'V16', 'V17', 'V18', 'V19', 'V20', 'V21', 'V22', 'V23', 'V24', 'V25', 'V26', 'V27', 'V28', 'V29', 'V30', 'V31', 'V32', 'V33', 'V34', 'V35', 'V36', 'V37', 'V38', 'V39', 'V40', 'V41', 'V42', 'V43'], d_columns=[], target='label')
2024-03-11 22:36:57,734 AutoFE.py:36 __init__ INFO | File name: SPECTF
2024-03-11 22:36:57,734 AutoFE.py:37 __init__ INFO | Data shape: (267, 45)
2024-03-11 22:36:58,214 AutoFE.py:90 set_cuda INFO | Use device: 0, cuda:0, NVIDIA GeForce RTX 4060 Laptop GPU
2024-03-11 22:36:58,214 AutoFE.py:80 __init__ INFO | Done AutoFE initialization.
2024-03-11 22:36:58,419 AutoFE.py:103 fit_attention INFO | score_b=0.7976939203354297, scores_b=[0.83333333 0.77777778 0.77358491 0.75471698 0.8490566 ]
2024-03-11 22:40:42,203 main_ppo.py:80 <module> INFO | Traceback (most recent call last):
  File "D:\software\anaconda\envs\pytorch\lib\site-packages\pandas\core\indexes\base.py", line 3802, in get_loc
    return self._engine.get_loc(casted_key)
  File "pandas\_libs\index.pyx", line 138, in pandas._libs.index.IndexEngine.get_loc
  File "pandas\_libs\index.pyx", line 165, in pandas._libs.index.IndexEngine.get_loc
  File "pandas\_libs\hashtable_class_helper.pxi", line 5745, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas\_libs\hashtable_class_helper.pxi", line 5753, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 0

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\python files\pythonProject3\PPO_transformer_lstm\main_ppo.py", line 76, in <module>
    autofe.fit_attention(args)
  File "D:\python files\pythonProject3\PPO_transformer_lstm\AutoFE.py", line 152, in fit_attention
    w = multiprocess_reward(args, worker, c_columns, d_columns, scores_b, mode,model, metric, x_d_onehot, df_t.values, df_d_labelencode)
  File "D:\python files\pythonProject3\PPO_transformer_lstm\feature_engineer\training_ops.py", line 81, in multiprocess_reward
    x_c = worker.features_c[step]
  File "D:\software\anaconda\envs\pytorch\lib\site-packages\pandas\core\frame.py", line 3807, in __getitem__
    indexer = self.columns.get_loc(key)
  File "D:\software\anaconda\envs\pytorch\lib\site-packages\pandas\core\indexes\base.py", line 3804, in get_loc
    raise KeyError(key) from err
KeyError: 0

2024-03-11 22:40:42,203 main_ppo.py:82 <module> INFO | Total cost time: 224.4687 s.
